### ISSUE TITLE
train(start_date = TRAIN_START_DATE,        end_date = TRAIN_END_DATE,       ticker_list = config_tickers.DOW_30_TICKER,        data_source = 'alpaca',       time_interval= '1D',        technical_indicator_list= INDICATORS,       drl_lib='elegantrl',        env=env,       model_name='ppo',       if_vix=False,        erl_params=ERL_PARAMS,       cwd='./papertrading_erl', #current_working_dir       API_KEY = API_KEY,        API_SECRET = API_SECRET,        API_BASE_URL = API_BASE_URL,       break_step=1e5)

### ISSUE THREAD

**Issue by valleysprings** [AI4Finance-Foundation/FinRL#889]
I means the FinRL/finrl/agents/elegantrl/models.py has been outdate.

![image](https://user-images.githubusercontent.com/57850191/218309961-8e4aef44-a121-45f8-a98d-c11cffbedf92.png)

env-stocktrading environment has been updated quickly and only stable-baseline3 agent is working fine right now.

**Comment by valleysprings** [comment]
And another issue is there are some dupilicated codes in preprocessor and data_processors (for example proceesor_yahoofinance and yahoodownloader + preprocessor) and only the latter one is working fine in env-stocktrading environment. Hope these code will get an overhaul soon. Thank you!

**Comment by valleysprings** [comment]
elegantRL needs this parameters to init, while stocktradingenv cannot provide these arguments, thus resulting some errors.

![image](https://user-images.githubusercontent.com/57850191/218313693-5d4ce2d7-f433-4c33-b6e0-56d1e0235f2e.png)

![image](https://user-images.githubusercontent.com/57850191/218313769-ebe00d23-8ff4-40a2-8a5e-fb9d9f3530fc.png)

**Comment by valleysprings** [comment]
for the record, rllib has the same issue.

**Comment by YSLIU627** [comment]
a suggestion is to check the importing function. You should import the environment env_stocktrading_np.py instead of env_stocking.py, since the env class defined in the former file has the 'config' attribute.

**Comment by valleysprings** [comment]
Yeah, i have tried that. But I encounter other errors regarding to elegantRL. (haven't tried rllib before :)
